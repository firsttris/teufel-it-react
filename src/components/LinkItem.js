import React, {Component} from 'react';
import {
    Link
} from 'react-router-dom'

export default class LinkItem extends Component {
    constructor() {
        super();
        this.state = {
            hover: false
        }
    }

    onMouseLeave() {
        this.setState({hover: false});
    }

    onMouseEnter() {
        this.setState({hover: true});
    }

    getLinkStyle() {
        const linkStyle = {
            display: 'block',
            color: 'white',
            textAlign: 'center',
            padding: '14px 16px',
            textDecoration: 'none'
        }
        if (this.props.focus) {
            return {...linkStyle, ...{backgroundColor: '#f00'}};
        }
        if (this.state.hover) {
            return {...linkStyle, ...{backgroundColor: 'white', color: 'black'}};
        }
        return linkStyle;
    }

    render() {
        return (
            <li style={{float: 'left'}} onMouseLeave={this.onMouseLeave.bind(this)}
                onMouseEnter={this.onMouseEnter.bind(this)}><Link style={this.getLinkStyle()}
                                                                  to={this.props.path}>{this.props.name}</Link></li>
        );
    }
}