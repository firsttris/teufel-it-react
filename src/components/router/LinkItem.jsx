import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './linkItem.css';
import PropTypes from 'prop-types';

class LinkItem extends Component {
  constructor() {
    super();
    this.state = {
      hover: false
    };
  }

  onMouseLeave() {
    this.setState({ hover: false });
  }

  onMouseEnter() {
    this.setState({ hover: true });
  }

  getLinkStyle() {
    const linkStyle = {
      display: 'block',
      color: 'white',
      textAlign: 'center',
      padding: '14px 16px',
      textDecoration: 'none'
    };
    if (this.props.focus) {
      return { ...linkStyle, ...{ backgroundColor: '#f00' } };
    }
    if (this.state.hover) {
      return { ...linkStyle, ...{ backgroundColor: 'white', color: 'black' } };
    }
    return linkStyle;
  }

  render() {
    return (
      <li
        className="link-item"
        onMouseLeave={this.onMouseLeave.bind(this)}
        onMouseEnter={this.onMouseEnter.bind(this)}
      >
        <Link style={this.getLinkStyle()} to={this.props.path}>
          {this.props.name}
        </Link>
      </li>
    );
  }
}

LinkItem.propTypes = {
  focus: PropTypes.bool,
  path: PropTypes.string,
  name: PropTypes.string
};

export default LinkItem;
