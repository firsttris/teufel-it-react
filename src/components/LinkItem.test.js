import React from 'react';
import renderer from 'react-test-renderer';
import LinkItem from "./LinkItem";
import {
  Router
} from 'react-router-dom'
import createHistory from 'history/createMemoryHistory'

test('FadeCard changes the class when hovered', () => {
  const component = renderer.create(
    <Router history={createHistory()}>
      <LinkItem focus={true} path="/" name="Entwicklung"></LinkItem>
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});