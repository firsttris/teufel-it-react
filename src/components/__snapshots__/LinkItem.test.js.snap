// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeCard changes the class when hovered 1`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "float": "left",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f00",
        "color": "white",
        "display": "block",
        "padding": "14px 16px",
        "textAlign": "center",
        "textDecoration": "none",
      }
    }
  >
    Entwicklung
  </a>
</li>
`;

exports[`FadeCard changes the class when hovered 2`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "float": "left",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f00",
        "color": "white",
        "display": "block",
        "padding": "14px 16px",
        "textAlign": "center",
        "textDecoration": "none",
      }
    }
  >
    Entwicklung
  </a>
</li>
`;

exports[`FadeCard changes the class when hovered 3`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "float": "left",
    }
  }
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#f00",
        "color": "white",
        "display": "block",
        "padding": "14px 16px",
        "textAlign": "center",
        "textDecoration": "none",
      }
    }
  >
    Entwicklung
  </a>
</li>
`;
